cargo-guppy/src/lib.rs

```diff
@@ -18,18 +25,136 @@ pub fn cmd_diff(json: bool, old: &str, new: &str) -> Result<(), Error> {
     Ok(())
 }
 
-pub fn cmd_count() -> Result<(), Error> {
+pub fn cmd_count() -> Result<(), anyhow::Error> {
     let lockfile = Lockfile::from_file("Cargo.lock")?;
 
     println!("Third-party Packages: {}", lockfile.third_party_packages());
 
     Ok(())
 }
 
-pub fn cmd_dups() -> Result<(), Error> {
+pub fn cmd_dups() -> Result<(), anyhow::Error> {
     let lockfile = Lockfile::from_file("Cargo.lock")?;
 
     lockfile.duplicate_packages();
 
     Ok(())
 }
+
+arg_enum! {
+    #[derive(Debug)]
+    pub enum Kind {
+        All,
+        Workspace,
+        ThirdParty,
+    }
+}
+
+struct NameVisitor;
+
+impl PackageDotVisitor for NameVisitor {
+    fn visit_package(&self, package: &PackageMetadata, mut f: DotWrite<'_, '_>) -> fmt::Result {
+        write!(f, "{}", package.name())
+    }
+
+    fn visit_link(&self, _link: DependencyLink<'_>, mut f: DotWrite<'_, '_>) -> fmt::Result {
+        write!(f, "")
+    }
+}
+
+pub fn cmd_select(
+    kind: Kind,
+    target: Option<String>,
+    output: Option<String>,
+    include_dev: bool,
+    include_build: bool,
+    omit_edges_into: &[String],
```

Comment:
Might be worth passing this in as an `impl IntoIterator<Item = String>` so you don't have to do an unnecessary `cloned()`. (Or if you can figure out how to have a single struct argument in structopt, just use that.)

Reply:
Went with the single struct argument.